omnigibson/object_states/object_state_base.py

```diff
@@ -357,7 +342,6 @@ def remove(self):
 
     def dump_state(self, serialized=False):
         assert self._initialized
-        assert self.stateful
```

Comment:
hmm interesting, why remove?

Reply:
This PR removes state size and the stateful as described in the description: `Remove state size assumption and allow each serializable instance to deserialize a variable number of items (will be used later for RigidPrim state, currently not included in PR due to some issues with it)`

Reply:
how does an object know whether an object is actually stateful?